components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/internal/database/DashboardMetadataDao.java

```diff
@@ -58,6 +57,61 @@ public DashboardMetadataDao(DataSource dataSource, QueryManager queryManager) {
         this.queryManager = queryManager;
     }
 
+    public void initDashboardTable() throws DashboardException {
+        if (!tableExists(QueryManager.DASHBOARD_RESOURCE_TABLE)) {
+            this.createDashboardResourceTable();
+        }
+    }
+
+    /**
+     * Create dashboard resource table.
+     */
+    public void createDashboardResourceTable() throws DashboardException {
```

Comment:
Does this needs to be public? If only `initDashboardTable` is called from outside, then this can be private.